spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -650,10 +649,12 @@ def main(argv: Sequence[str]):
             printv(f'\n{metric} ratio norm = {metric_ratio_norm_result}', verbose=verbose, type='info')
             printv(f'\n{metric} ratio PAM50 = {metric_ratio_PAM50_result}', verbose=verbose, type='info')
 
-    df_metric_ratios = pd.DataFrame(rows, columns=['filename', 'compression_level', 'slice(I->S)',
-                                                   f'{arguments.metric}_ratio',
-                                                   f'{arguments.metric}_ratio_PAM50',
-                                                   f'{arguments.metric}_ratio_PAM50_normalized'])
+    metric_columns = [
+        f'{arguments.metric}_ratio',
+        f'{arguments.metric}_ratio_PAM50',
+        f'{arguments.metric}_ratio_PAM50_normalized',
+    ]
+    df_metric_ratios = pd.DataFrame.from_records(rows, index=INDEX_COLUMNS, columns=INDEX_COLUMNS + metric_columns)
```

Comment:
I was curious what the difference was between `.from_records()` and simply using the `pd.Dataframe` constructor.

From what I can tell, `from_records()` allows us to select existing columns a a `MultiIndex`? (Whereas `pd.Dataframe`'s `index` argument functions differently -- it's meant to specify the actual index values for each row. Meaning, we'd have to make our own `MultiIndex` and pass it to the constructor.)

`from_records()` seems much simpler/cleaner, so :+1:  from me. :)

Reply:
That's it exactly, yes.